controller/service.go

```diff
@@ -195,22 +213,34 @@ func (c *controller) allocateIPs(key string, svc *v1.Service) ([]net.IP, error)
 
 	// If the user asked for a specific IPs, try that.
 	if len(desiredLbIPs) > 0 {
-		if serviceIPFamily != desiredLbIPFamily {
-			return nil, fmt.Errorf("requested loadBalancer IP(s) %q does not match the ipFamily of the service", desiredLbIPs)
-		}
-		if err := c.ips.Assign(key, desiredLbIPs, k8salloc.Ports(svc), k8salloc.SharingKey(svc), k8salloc.BackendKey(svc)); err != nil {
+		desiredPool, err := c.ips.GetPoolOwner(desiredLbIPs)
```

Comment:
same here. I'd keep the existing logic and eventually extend. If we got this far, we already did the pool compatibility checks?